plugin/trino-neo4j/src/main/java/io/trino/plugin/neo4j/Neo4jClient.java

```diff
@@ -0,0 +1,339 @@
+import io.trino.spi.connector.SchemaTableName;
+import org.neo4j.cypherdsl.core.AliasedExpression;
+import org.neo4j.cypherdsl.core.Cypher;
+import org.neo4j.cypherdsl.core.Node;
+import org.neo4j.cypherdsl.core.Relationship;
+import org.neo4j.cypherdsl.core.Statement;
+import org.neo4j.driver.AuthToken;
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Driver;
+import org.neo4j.driver.GraphDatabase;
+import org.neo4j.driver.Session;
+import org.neo4j.driver.SessionConfig;
+import org.neo4j.driver.Value;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.concurrent.ExecutionException;
+
+import static com.google.common.base.Throwables.throwIfInstanceOf;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.neo4j.Neo4jNamedRelationHandle.TableType.NODE;
+import static io.trino.spi.StandardErrorCode.TABLE_NOT_FOUND;
+import static java.util.Locale.ENGLISH;
+
+public class Neo4jClient
+        implements AutoCloseable
+{
+    private static final Logger log = Logger.get(Neo4jClient.class);
+
+    private static final String SYSTEM_DATABASE = "system";
+    private static final String DEFAULT_DATABASE = "neo4j";
+
+    private final Driver driver;
+    private final Neo4jTypeManager typeManager;
+
+    private final Cache<SchemaTableName, Neo4jRemoteTableName> tableNameCache;
+    private final Cache<SchemaTableName, Neo4jTable> tableCache;
+
+    @Inject
+    public Neo4jClient(
+            Neo4jConnectorConfig config,
+            Neo4jTypeManager typeManager)
+    {
+        this.driver = GraphDatabase.driver(config.getURI(), getAuthToken(config));
+        this.typeManager = typeManager;
+        this.tableNameCache = EvictableCacheBuilder.newBuilder()
+                .expireAfterWrite(Duration.ofMinutes(1))
+                .build();
+
+        this.tableCache = EvictableCacheBuilder.newBuilder()
+                .expireAfterWrite(Duration.ofMinutes(1))
+                .build();
+    }
+
+    private static AuthToken getAuthToken(Neo4jConnectorConfig config)
+    {
+        return switch (config.getAuthType().toLowerCase(ENGLISH)) {
+            case "basic" -> AuthTokens.basic(config.getBasicAuthUser(), config.getBasicAuthPassword());
+            case "bearer" -> AuthTokens.bearer(config.getBearerAuthToken());
+            case "none" -> AuthTokens.none();
+            default -> throw new ConfigurationException(ImmutableList.of(new Message("Unknown neo4j.auth.type: %s".formatted(config.getAuthType()))));
+        };
+    }
+
+    public List<String> listSchemaNames(ConnectorSession connectorSession)
+    {
+        try (Session session = this.driver.session(SessionConfig.forDatabase(SYSTEM_DATABASE))) {
+            return session.executeRead(tx -> {
+                return tx.run("SHOW DATABASES YIELD name, type WHERE type <> 'system' RETURN distinct(name)")
```

Comment:
I had to change this query to make it work with the latest version of Trino:
`"SHOW DATABASES YIELD name WHERE name <> 'system' RETURN distinct(name)"`
Apparently, SHOW DATABASES no longer returns type.